http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java

```diff
@@ -524,6 +524,10 @@ ExecutionFlow<MutableHttpResponse<?>> createResponseForBody(HttpRequest<?> reque
                 }
                 response.body(null);
             }
+            if (response.getContentType().isEmpty()) {
+                MediaType mediaType = resolveDefaultResponseContentType(request, routeInfo);
+                response.contentType(mediaType);
+            }
```

Comment:
not sure about this change, why is this necessary?

Reply:
don't think the core Router should be changed for this, btw there are already implementation of the routing for serverless that are based on the Servlet module like https://github.com/micronaut-projects/micronaut-gcp/tree/master/gcp-function-http